ultravox/inference/infer.py

```diff
@@ -30,6 +32,60 @@ def __init__(
         self.processor = processor
         self.dtype = dtype
 
+    def batch_infer(
+        self,
+        all_samples: List[datasets.VoiceSample],
+        batch_size: int = 1,
+        max_tokens: Optional[int] = None,
+        temperature: Optional[float] = None,
+    ) -> base.InferenceGenerator:
+        data_collator = datasets.DataCollatorForSeq2SeqWithAudio(
+            tokenizer=self.tokenizer,
+            include_alt_fields=False,
+        )
+        sample_additional_info = []
+        for i, sample in enumerate(all_samples):
+            question_text = sample.audio_transcript
+            expected_answer = sample.messages[-1]["content"]
+            sample_additional_info.append(
+                {
+                    "index": i,
+                    "question_text": question_text,
+                    "expected_answer": expected_answer,
+                }
+            )
+            sample.messages = sample.messages[:-1]
+
+        dataset: Dataset[Any] = Dataset.from_list(
+            [self._dataproc(s, batch=True) for s in all_samples]
+        )
```

Comment:
Similarly, applying `self._dataproc` to all samples at once will not scale for a large data set. Take a look at  `train.py` on how data processor in used in datasets and in batching. I think data processor should be applied to one batch at a time.